drivers: gpio: nrfx: Add support for GPIOTE0 on nrf54h20 cpurad

nordic-krch opened the pull request:
Add support for special GPIOTE0 instance on nrf54h20/cpurad.

This instance requires special handling because:
  - there is no support for PORT event (level interrupts)
  - TE channels are fixed to the pin
  - pins are configured through CTRLSEL (new nrf-regtool version needed)
  - new nrfx_gpiote driver needed

~~`DNM` until `nrf-regtool` v9.2.0 is available.~~

nordic-krch committed:
tests: drivers: gpio: basic_api: Improve test

Add return code checking for gpio_pin_interrupt_configure functions.
Add disabling pin interrupt at the end of the test to restore pin
state.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
dts: vendor: nordic: nrf54h20: Add gpiote0 node

Add GPIOTE0 instance in radio peripherals.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
dts: bindings: gpio: nrf-gpiote: Extend description

Add new feature flags to gpiote node.
Include pinctrl. Pins used by GPIOTE0 on nrf54h20/cpurad require CTRLSEL
configuration. Pins are listed using pinctrl and parsed by nrf-regtool
to prepare UICR configuration.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
drivers: gpio: nrfx: Add support for GPIOTE0 on cpurad

Add support for special GPIOTE0 instance on nrf54h20/cpurad.
This instance requires special handling because:
- there is no support for PORT event (level interrupts)
- TE channels are fixed to the pin

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
modules: hal_nordic: nrfx: Add GPIOTE config flag

Add NRFX_GPIOTE_VAR_FEATURE_SUPPORT feature flag to nrfx configuration.
It enables support for special GPIOTE0 instance in nrfx_gpiote driver.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: drivers: gpio: basic_api: Add nrf54h20/cpurad GPIOTE0 config

Add configuration for testing GPIOTE0 instance on nrf54h20/cpurad.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

fabiobaltieri commented:
Hey, this needs a rebase and force push to pick up the CI error workaround.

nordic-krch committed:
modules: hal_nordic: Bump nrf-regtool to 9.2.0

Bump required version of nrf-regtool to 9.2.0.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

kl-cruz reviewed (dismissed):
Please set hal_nordic hash reference if possible

bjarki-andreasen commented:
Rebased

bjarki-andreasen-nordic committed:
soc: nordic: fix validate_base_addresses NRF_GPIOTE0 for radiocore

Map definition of NRF_GPIOTE0 to NRF_RADIOCORE_GPIOTE when MDK
defines NRF_RADIOCORE_GPIOTE instead of NRF_GPIOTE0

Signed-off-by: Bjarki Arge Andreasen <bjarki.andreasen@nordicsemi.no>

bjarki-andreasen commented on dts/vendor/nordic/nrf54h20.dtsi:
```diff
@@ -412,6 +412,16 @@
 				interrupts = <37 NRF_DEFAULT_IRQ_PRIORITY>;
 			};
 
+			gpiote0: gpiote@27000 {
+				compatible = "nordic,nrf-gpiote";
+				reg = <0x27000 0x1000>;
+				status = "disabled";
+				interrupts = <39 NRF_DEFAULT_IRQ_PRIORITY>;
+				instance = <0>;
+				no-port-event;
+				fixed-channels-supported;
+			};
```
@nordic-krch is on vacation, I have pushed the suggestion
